plugins/agents/filesystem/Parser.go

```diff
@@ -0,0 +1,67 @@
+package filesystem
+
+import (
+	"bytes"
+	"strconv"
+	"strings"
+)
+
+type commandParser interface {
+	parse([]byte) ([]filesystemInfo, error)
+}
+
+type dfCommandParser struct {
+}
+
+func (p *dfCommandParser) parse(theOutput []byte) ([]filesystemInfo, error) {
+	lines := bytes.Split(theOutput, []byte("\n"))
+	filesystems := make([]filesystemInfo, 0)
+
+	for _, line := range lines {
+		newFsInfo, newFsErr := p.parseLine(line)
+		if newFsErr != nil {
+			return nil, newFsErr
+		}
+		if newFsInfo == nil {
+			continue
+		}
+		filesystems = append(filesystems, *newFsInfo)
+	}
+	return filesystems, nil
+}
+
+func (p *dfCommandParser) parseLine(singleLine []byte) (*filesystemInfo, error) {
+	if len(singleLine) == 0 || singleLine[0] != byte('/') {
```

Comment:
Maybe this deserves a comment. I assume it's to remove `tmpfs`, NFS-mounted volumes etc, but it's not obvious.